src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperationService.cs

```diff
@@ -47,11 +47,19 @@ private async Task ExecuteEgressOperation(EgressRequest egressRequest, Cancellat
                 CancellationToken token = linkedTokenSource.Token;
                 token.ThrowIfCancellationRequested();
 
-                var result = await egressRequest.EgressOperation.ExecuteAsync(_serviceProvider, token);
+                try
+                {
+                    var result = await egressRequest.EgressOperation.ExecuteAsync(_serviceProvider, token);
 
-                //It is possible that this operation never completes, due to infinite duration operations.
-
-                _operationsStore.CompleteOperation(egressRequest.OperationId, result);
+                    //It is possible that this operation never completes, due to infinite duration operations.
+                    _operationsStore.CompleteOperation(egressRequest.OperationId, result);
+                }
+                //This is unexpected, but an unhandled exception should still fail the operation.
+                catch (Exception e)
+                {
+                    _operationsStore.CompleteOperation(egressRequest.OperationId, ExecutionResult<EgressResult>.Failed(e));
+                    throw;
```

Comment:
I think this should not throw; this will cause an UnobservedTaskException to occur. However, I don't think faulting is enabled in dotnet-monitor when a task exception is unobserved.